keps/sig-api-machinery/4006-transition-spdy-to-websockets/README.md

```diff
@@ -417,14 +484,40 @@ affect the perceived performance.
 3. As releases increment, the probablity of a WebSocket enabled `kubectl` communicating
 with an older non-WebSocket enabled API Server decreases.
 
-### Proposal: `K8s-Websocket-Protocol: stream-translate` Header
-
-In addition to the current SPDY-based clients, there are other current WebSocket clients,
-including a javascript/browser-based client. In order to distinguish these older
-WebSocket clients from the new stream-translated WebSocket clients, we propose adding
-a new header `K8s-Websocket-Protocol: stream-translate`. As described further in the
-next `Proposal` section, this header allows newer clients to delegate to the
-`StreamTranslatorProxy` to translate WebSockets data messages to SPDY.
+### Proposal: New `RemoteCommand` Sub-Protocol Version - `v5.channel.k8s.io`
+
+The latest RemoteCommand version does not address an important protocol feature--a
+stream `CLOSE` signal. In order to communicate to another endpoint that the current
+stream of sending data is complete, a `CLOSE` signal is necessary. This problem
+currently arises when sending data over the STDIN stream, and it is more fully described
+in the following issue: [exec over web sockets protocol is flawed](https://github.com/kubernetes/kubernetes/issues/89899).
+A new RemoteCommand version (`v5.channel.k8s.io`) adds this `CLOSE` signal.
+
+### Proposal: New `PortForward` Sub-Protocol Version - `v2.portforward.k8s.io`
+
+We propose a new PortForward version `v2.portforward.k8s.io`, which would add two
+stream `CLOSE` signals. The first `CLOSE` signal would communicate to the server
+that the writing from the local port has completed. The second `CLOSE` signal
+would communicate from the server to the local client that the resulting data
+sent back to the client has completed. In order to properly identify and
+separate different (possibly simultaneous) streams, this new protocol will
+contain a mechanism to dynamically create unique stream ids for each request
+(regardless of remote port).
+
+### Proposal: New WebSocket Stream Identifier Generation Mechanism
```

Comment:
Since port number is carried via header in SPDY, there shouldn't be any port number limit(or client's limitation is applied). 

But there was a limit in previous websocket attempt https://github.com/kubernetes/kubernetes/pull/50428/files#diff-aea43143d31d9ec5cce18515c033e2f4f0f797fb7221318811cd75f20ee2c301R306-R315 which is 32,767 and I think, we will have too and it's worth to mention this. WDYT?.